pkg/daemon/ceph/client/pool.go

```diff
@@ -183,11 +183,27 @@ func CreatePool(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec
 }
 
 func CreatePoolWithPGs(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec, appName, pgCount string) error {
+	var autoscaleMode string
```

Comment:
How about just assign this by default first, then no need for lines 202-204.
```suggestion
	autoscaleMode := pool.AutoscaleMode
```